src/Functions/HasTokenImpl.h

```diff
@@ -24,14 +24,15 @@ struct HasTokenImpl
     static constexpr bool supports_start_pos = false;
     static constexpr auto name = Name::name;
 
-    static ColumnNumbers getArgumentsThatAreAlwaysConstant() { return {1, 2};}
+    static ColumnNumbers getArgumentsThatAreAlwaysConstant() { return {1, 2}; }
 
     static void vectorConstant(
         const ColumnString::Chars & haystack_data,
         const ColumnString::Offsets & haystack_offsets,
         const std::string & pattern,
         const ColumnPtr & start_pos,
-        PaddedPODArray<UInt8> & res)
+        PaddedPODArray<UInt8> & res,
+        ColumnUInt8 * null_map = nullptr)
```

Comment:
It would be more uniform and have less potential for surprises if null_map was mandatory (non-optional). Please also name it `res_null` such that it looks like an output variable.